### PR TITLE
feature/AKD-190: Added Google BigQuery Dataset All Users Policy plugin and test cases

### PR DESCRIPTION
**Pull request by AkhtarAmir**

### CODE REVIEW

File: helpers/google/index.js [hunk 1]
```diff
@@ -348,14 +350,15 @@ var execute = function(LocalGoogleConfig, collection, service, callObj, callKey,
...
     var parentParams;
+    if (callObj.project) callObj.params.projectId = LocalGoogleConfig.project;
     if (callObj.nested && callObj.parent) {
         parentParams = {auth: callObj.params.auth, parent: callObj.params.parent};
         executor['projects']['locations'][service][callKey](parentParams, LocalGoogleConfig, executorCb);
     } else if (callObj.nested) {
         parentParams = {auth: callObj.params.auth, parent: callObj.params.parent};
         executor['projects']['locations']['keyRings'][service][callKey](parentParams, LocalGoogleConfig, executorCb);
     } else if (callObj.resource) {
-        parentParams = {auth: callObj.auth, resource_: LocalGoogleConfig.project};
```
Comment (giorod3) [inline helpers/google/index.js:358]: check this

**Review by giorod3 — CHANGES_REQUESTED** [review]
everything looks good except the one comment

File: helpers/google/index.js (hunk 1 above)
Comment (AkhtarAmir) [reply helpers/google/index.js]: fixed

File: plugins/google/bigquery/datasetAllUsersPolicy.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            }
+
+            async.each(datasetsGet.data, (dataset, dcb) => {
+                if (!dataset.id) return dcb();
+
+                var permissionStr = [];
+                if (dataset.access) {
+                    for (let rolePermission of dataset.access) {
+                        for (let property in rolePermission) {
+                            if (!rolePermission['role']) continue;
```
Comment (giorod3) [inline plugins/google/bigquery/datasetAllUsersPolicy.js:41]:
```suggestion
                    for (let rolePermission of dataset.access) {
                        if (!rolePermission['role']) continue;
                        for (let property in rolePermission) {
```

**Pull request merged**